### DISCUSSION TITLE
How to setup image permission to public

### DISCUSSION THREAD

**Discussion by gpb3037** [earthlab-education/Earth-Analytics-AY24#89] [Q&A]
I created a .img directory in my portfolio by making a .keep file and uploaded a .jpg image
I can see it when I run the code but can't see it from the class distribution
Can't seem to figure out how to establish permissions for others to see it

Code:
 <img src="https://github.com/gpb3037/gpb3037.github.io/blob/87f905ddab695eb5087e20aab6cbd195a71fb6a8/img/gpb-pi-777wm.jpg" alt="piFspiral" width="200" height="200" > 
 
I also tried putting it elsewhere but this did not work either
https://o365coloradoedu-my.sharepoint.com/:i:/g/personal/geba2669_colorado_edu/EZPa80ruU7FNqO7Cxh7TV68BRvXothAfLHPJLBwi5QcQgA

**Comment by eculler** [comment]
Hi @gpb3037, great question! I think you are actually having a similar problem with both of your links, in that your image source is a page about your image rather than the image itself. Notice that when you follow [this link from your index.md file](https://github.com/gpb3037/gpb3037.github.io/blob/87f905ddab695eb5087e20aab6cbd195a71fb6a8/img/gpb-pi-777wm.jpg) you actually end up at a GitHub page that does have your image on it, but also a lot of other stuff. I believe OneDrive links aren't really meant to host image file types for the internet, and you can see if you follow that link that you're actually loading an image viewer rather than the image itself. If you're using hosting services separate from GitHub, you'll usually want to keep an eye out for an "embed" link, which is intended for including content in other webpages. You can sometimes get there also by right-clicking and selecting "Open Image in New Tab" or something like that. For example, [this is the link I got for your image on GitHub that is just for the image](https://raw.githubusercontent.com/gpb3037/gpb3037.github.io/87f905ddab695eb5087e20aab6cbd195a71fb6a8/img/gpb-pi-777wm.jpg), using that method.

Luckily, you already did the work of uploading your images, which means you can put in a _relative_ link (to where the image is in your repository e.g. img/piFspiral.jpg) rather than an _absolute_ link (to somewhere on the broader internet, e.g. https://www...). I'd go with that for now -- we can talk other ways of hosting images down the line if you run in to something to large to upload directly to your repository.

**Reply by BioNCphilic** [reply]
I'm following this as a guide to complete my portfolio assignment, but I get an error code when I try to access my portfolio link: BioNCphilic.GitHub.io. I'm not sure what to do next.

**Reply by BioNCphilic** [reply]
Never mind - I was confused. I didn't realized I was the one to create the repository - thought the instructor was setting that up linked to the course. I think my portfolio is now active.
